src/components/program/ProgramPage.jsx

```diff
@@ -8,45 +6,27 @@ import { ErrorPage } from '@edx/frontend-platform/react';
 
 import { MainContent, Sidebar } from '../layout';
 import { LoadingSpinner } from '../loading-spinner';
-import { ProgramContextProvider } from './ProgramContextProvider';
 import ProgramHeader from './ProgramHeader';
 import ProgramMainContent from './ProgramMainContent';
 import ProgramSidebar from './ProgramSidebar';
 
-import { useAllProgramData } from './data/hooks';
 import ProgramEndorsements from './ProgramEndorsements';
 import ProgramFAQ from './ProgramFAQ';
 import ProgramCTA from './ProgramCTA';
 import NotFoundPage from '../NotFoundPage';
 import { PROGRAM_NOT_FOUND_MESSAGE, PROGRAM_NOT_FOUND_TITLE } from './data/constants';
 import ProgramDataBar from './ProgramDataBar';
-import { useEnterpriseCustomer } from '../app/data';
+import { useEnterpriseCustomer, useProgramDetails } from '../app/data';
 
 const ProgramPage = () => {
-  const { programUuid } = useParams();
   const { data: enterpriseCustomer } = useEnterpriseCustomer();
-
-  const [programData, fetchError] = useAllProgramData({ enterpriseUuid: enterpriseCustomer.uuid, programUuid });
-
-  const initialState = useMemo(
-    () => {
-      if (!programData) {
-        return undefined;
-      }
-      const { programDetails } = programData;
-
-      return {
-        program: programDetails,
-      };
-    },
-    [programData],
-  );
+  const { data: program, isError: fetchError, isLoading } = useProgramDetails();
 
   if (fetchError) {
     return <ErrorPage message={fetchError.message} />;
   }
 
-  if (!initialState) {
+  if (isLoading) {
```

Comment:
If data is pre-seeded in the query cache, there's likely no need for a loading state in this component since `isLoading` would generally always be `false`?